latex/epoch.tex

```diff
@@ -1,54 +1,614 @@
+  \emph{UTxOE transitions}
   \begin{equation*}
-    \_ \vdash \_ \trans{poolreap}{} \_ \in
-    \powerset (\Slot \times \PState \times \PState)
+    \_ \vdash
+    \var{\_} \trans{utxoep}{} \var{\_}
+    \subseteq \powerset (\UTxOEpEnv \times \UTxOState \times \UTxOState)
   \end{equation*}
   %
-  \caption{Pool Reap Transition}
-  \label{fig:ts-types:pool-reap}
+  \caption{UTxO Epoch transition-system types}
+  \label{fig:ts-types:utxoe}
 \end{figure}
 
+%%
+%% Figure - UTxO Epoch Rule
+%%
+\begin{figure}
+  \begin{equation}\label{eq:utxoep}
+    \inference[UTxO-epoch]
+    {
+    }
+    {
+      \begin{array}{l}
+        \var{slot}\\
+        \var{pc}\\
+        \var{stkeys}\\
+        \var{stpools}\\
+      \end{array}
+      \vdash
+      \left(
+        \begin{array}{r}
+          \var{utxo} \\
+          \var{deposits} \\
+          \var{fees} \\
+          \var{wdrls} \\
+        \end{array}
+      \right)
+      \trans{utxoep}{}
+      \left(
+        \begin{array}{r}
+          \var{utxo} \cup \var{wdrls} \\
+          \obligation{pc}{stkeys}{stpools}{slot} \\
+          0 \\
+          \emptyset \\
+        \end{array}
+      \right)
+    }
+  \end{equation}
+  \caption{UTxO Epoch inference rules}
+  \label{fig:rules:utxoep}
+\end{figure}
 
 %%
-%% Figure - Pool Rules
+%% Figure - Accounting Defs
 %%
 \begin{figure}
+  \emph{Accounting Fields}
+  \begin{equation*}
+    \Accnt =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{treasury} & \Coin & \text{treasury pool}\\
+        \var{reserves} & \Coin & \text{reserve pool}\\
+        \var{rewardPool} & \Coin & \text{reward pool}\\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{Accounting environment}
+  \begin{equation*}
+    \AccntEnv =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{slot} & \Slot & \text{current slot}\\
+        \var{pc} & \PrtclConsts & \text{protocol constants}\\
+        \var{production} & \Production & \text{blocks produced in the epoch}\\
+        \var{utxoSt} & \UTxOState & \text{utxo state}\\
+        \var{pstate} & \PState & \text{pool state}\\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{Accounting States}
+  \begin{equation*}
+    \AccntState =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{accnt} & \Accnt & \text{accounting}\\
+        \var{dstate} & \DState & \text{delegation state}\\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{Accounting transitions}
+  \begin{equation*}
+    \_ \vdash
+    \var{\_} \trans{accnt}{} \var{\_}
+    \subseteq \powerset (\AccntEnv \times \AccntState \times \AccntState)
+  \end{equation*}
+  %
+  \caption{Accounting transition-system types}
+  \label{fig:ts-types:accnt}
+\end{figure}
 
-  \begin{equation}\label{eq:pool-reap}
-    \inference[Pool-Reap]
+%%
+%% Figure - Accounting Rules
+%%
+\begin{figure}
+  \begin{equation}\label{eq:accnt}
+    \inference[Accounting]
+    {
+      \var{obl} = \obligation{pc}{stkeys}{stpools}{slot} \\
```

Comment:
The alignment of the `semantic` packages is awful when we have this multi-line inference rules. We might want to experiment with `array` or some other environment that allows a format that is easier on the eye.

I had the same problem, and this forced me to either introduce auxiliary functions, or split the rules in sub-rules to model orthogonal aspects.

Reply:
yea, I've been having a hard time keeping these looking ok. thanks for the tips!

Reply:
oh yea, using an array here is wonderful!